components/Common/DarkModeToggle/__tests__/index.test.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import DarkModeToggle from '../index';
+import { IntlProvider } from 'react-intl';
+
+let mockCurrentTheme = '';
+
+const mockToggleTheme = jest.fn().mockImplementation(() => {
+  mockCurrentTheme = mockCurrentTheme === 'dark' ? 'light' : 'dark';
+});
+
+// // mock dark mode module for controlling dark mode HOC behavior
+jest.mock('next-themes', () => ({
+  useTheme: () => {
+    return { theme: mockCurrentTheme, setTheme: mockToggleTheme };
+  },
+}));
+
+describe('DarkModeToggle Component', () => {
+  it('render dark mode toggle', () => {
+    const { container } = render(
+      <IntlProvider locale="en">
```

Comment:
We also need the `onError={() => {}}` stuff here 👀

Reply:
oops! will add it